validator/client/validator.go

```diff
@@ -931,6 +944,82 @@ func (v *validator) logDuties(slot types.Slot, duties []*ethpb.DutiesResponse_Du
 	}
 }
 
+// PrepareBeaconProposer calls the prepareBeaconProposer RPC to set the proposer information such as fee recipient.
+func (v *validator) PrepareBeaconProposer(ctx context.Context, km keymanager.IKeymanager) error {
+	if km == nil {
+		return errors.New("keymanager is nil when calling PrepareBeaconProposer")
+	}
+	pubkeys, err := km.FetchValidatingPublicKeys(ctx)
+	if err != nil {
+		return err
+	}
+	feeRecipients, err := v.feeRecipients(ctx, pubkeys)
+	if err != nil {
+		return err
+	}
+	if len(feeRecipients) == 0 {
+		log.Warnf("no valid validator indices were found, prepare beacon proposer request fee recipients array is empty")
+		return nil
+	}
+	if _, err := v.validatorClient.PrepareBeaconProposer(ctx, &ethpb.PrepareBeaconProposerRequest{
+		Recipients: feeRecipients,
+	}); err != nil {
+		return err
+	}
+	log.Infoln("Successfully prepared beacon proposer with fee recipient to validator index mapping.")
+	return nil
+}
+
+func (v *validator) feeRecipients(ctx context.Context, pubkeys [][fieldparams.BLSPubkeyLength]byte) ([]*ethpb.PrepareBeaconProposerRequest_FeeRecipientContainer, error) {
+	var validatorToFeeRecipientArray []*ethpb.PrepareBeaconProposerRequest_FeeRecipientContainer
+	// need to check for pubkey to validator index mappings
+	for _, key := range pubkeys {
+		feeRecipient := common.HexToAddress(fieldparams.EthBurnAddressHex)
```

Comment:
Move this to L992 and change it to `v.prepareBeaconProposalConfig.DefaultConfig.FeeRecipient` as default. Then you can get rid of the else to just
```go
if option != nil {
				feeRecipient = option.FeeRecipient
			}
```